src/Microsoft.DotNet.Build.CloudTestTasks/UploadClient.cs

```diff
@@ -114,7 +113,11 @@ public string EncodeBlockIds(int numberOfBlocks, int lengthOfId)
                                 req.Content = contentStream;
                                 using (HttpResponseMessage response = await client.SendAsync(req, ct))
                                 {
-                                    this.log.LogMessage(
+                                    if (!response.IsSuccessStatusCode)
```

Comment:
It would be handled by the caller (see UploadToAzure task).  I admit I'm not familiar with the async pattern, is there a better way I should be returned a failed task?